webauthn4j-core/src/main/java/com/webauthn4j/data/payment/PaymentCurrencyAmount.java

```diff
@@ -0,0 +1,51 @@
+package com.webauthn4j.data.payment;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.webauthn4j.util.AssertUtil;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public class PaymentCurrencyAmount implements Serializable {
+    private final String currency;
+    private final String value;
+
+    @JsonCreator
+    public PaymentCurrencyAmount(@NonNull @JsonProperty("currency") String currency,
+                                 @NonNull @JsonProperty("value") String value) {
+        AssertUtil.notNull(currency, "payment amount currency must not be null");
+        AssertUtil.notNull(value, "payment amount value must not be null");
+        this.currency = currency;
+        this.value = value;
+    }
+
+    public @NonNull String getCurrency() {
+        return currency;
+    }
+
+    public @NonNull String getValue() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        PaymentCurrencyAmount that = (PaymentCurrencyAmount) o;
+
+        if (!currency.equals(that.currency)) return false;
+
+        try {
+            return Double.parseDouble(value) == Double.parseDouble(that.value);
+        } catch (NumberFormatException e) {
+            return false;
+        }
```

Comment:
This makes following code false:
```
new PaymentCurrencyAmount("USD","invalid").equals(new PaymentCurrencyAmount("USD","invalid"))
```
I think it is inappropriate. Verification should be conducted at the designated place (`AuthenticationDataValidator#validatePaymentAmount`).

Reply:
amount should be a valid number, so your example should return false imo. Alternatively we can throw a numberformat exception to handle this case?

Reply:
I think `new PaymentCurrencyAmount("USD","invalid").equals(new PaymentCurrencyAmount("USD","invalid"))` should return `true`. 
Even though the value is invalid from the point of `PaymentCurrencyAmount` design, `equals` method need to just focus on value equality.

Reply:
ok i see your point, i will then extract the extra logic from the `equals` to the `AuthenticationDataValidator#validatePaymentAmount` method